sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDiagnosticsContext.java

```diff
@@ -556,10 +581,233 @@ public String toJson() {
                 return snapshot;
             }
 
+            this.systemUsage = ClientSideRequestStatistics.fetchSystemInformation();
             return this.cachedRequestDiagnostics = getRequestDiagnostics();
         }
     }
 
+    /**
+     * Gets the UserAgent header value used by the client issueing this operation
+     * NOTE: this information is not included in the json representation returned from {@link #toJson()} because it
+     * is usually only relevant when thresholds are violated, in which case the entire diagnostics json-string is
+     * included. Calling this method will lazily collect the user agent - which can be useful when writing
```

Comment:
nit: comment: since we have changed the implementation, so it is not lazily calculated anymore?

Reply:
Fixed in next iteration